### PR TITLE
only set MinSafeTS to 0 when all stores' are 0

### PR DESCRIPTION
**Pull request by zyguan**
https://github.com/tikv/client-go/pull/1277 introduce a regression: there might be a store whose safe ts is 0 for a long time, in this case, flashback will be blocked (ref: https://github.com/tikv/tikv/issues/13675). We should `setMinSafeTS(0)` only when all (NOT any) of `store.safe_ts = 0`.

### CODE REVIEW

File: tikv/kv.go [hunk 1]
```diff
@@ -646,13 +637,17 @@ func (s *KVStore) updateMinSafeTS(txnScope string, storeIDs []uint64) {
...
+			if safeTS != 0 && safeTS < minSafeTS {
 				minSafeTS = safeTS
 			}
 		} else {
 			minSafeTS = 0
 		}
 	}
+	// if minSafeTS is still math.MaxUint64, that means all store safe ts are 0, then we set minSafeTS to 0.
+	if minSafeTS == math.MaxUint64 {
+		minSafeTS = 0
```
Comment (cfzjywxk) [inline tikv/kv.go:649]: Or could we set it to something like `safepoint` ts ?

Comment (zyguan) [reply tikv/kv.go:649]: Sound reasonable, let me think about if it would break some constraints.

**Review by Defined2014 — APPROVED** [review]

File: internal/logutil/log.go [hunk 2]
```diff
@@ -60,3 +61,11 @@ type ctxLogKeyType struct{}
 // CtxLogKey is the key to retrieve logger from context.
 // It can be assigned to another value.
 var CtxLogKey interface{} = ctxLogKeyType{}
+
+// AssertWarn panics when in testing mode, and logs a warning msg otherwise.
+func AssertWarn(logger *zap.Logger, msg string, fields ...zap.Field) {
+	if testing.Testing() {
```
Comment (you06) [inline internal/logutil/log.go:67]: This condition is never "true" when running with bazel.
It's ok if it's used in client-go's test only.

Comment (zyguan) [reply internal/logutil/log.go:67]: Interesting, I tired to compile the following code by `go test -c ...` and run the binary, it also showed true.
```go
func TestDemo(t *testing.T) {
	fmt.Println(">>> in test?", testing.Testing())
}
```

Although the function is in internal package, I want it works in tidb ut. I'll figure out how bazel build/run test binaries.

Comment (you06) [reply internal/logutil/log.go:67]: FYI the only work way I find is TiDB pass the `intest.InTest` to client-go via an interface, not zero cost, and greatly impact the the client's interface.
If you use intest flag condition build in client-go, bazel compile will fail...

Comment (zyguan) [reply internal/logutil/log.go:67]: https://github.com/bazelbuild/rules_go/issues/3686 I realize the issue has be reported by @tiancaiamao . Maybe just let it only works in client-go ut by now.

Comment (you06) [reply internal/logutil/log.go:67]: LGTM

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**